spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/Merge.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.keras
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.nn.{CAddTable, CAveTable, CMaxTable, CMulTable, CosineDistance, DotProduct, JoinTable, ParallelTable, Sequential => TSequential}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{MultiShape, Shape}
+
+import scala.reflect.ClassTag
+
+class Merge[T: ClassTag](
+   val layers: Array[AbstractModule[Activity, Activity, T]] = null,
+   val mode: String = "sum",
+   val concatAxis: Int = -1,
+   var inputShape: MultiShape = null)(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Tensor[T], Tensor[T], T](KerasLayer.addBatch(inputShape)) {
+
+  private val mergeMode = mode.toLowerCase()
+  private var axis = concatAxis
+
+  require(mergeMode == "sum" || mergeMode == "mul" || mergeMode == "concat" || mergeMode == "ave"
+  || mergeMode == "cos" || mergeMode == "dot" || mergeMode == "max",
+  s"Invalid merge mode: $mergeMode")
+  require(layers.length >= 2, s"Merge must have at least two layers but found ${layers.length}")
+
+  override def getInputShape(): Shape = {
+    inputShape = if (inputShape != null) inputShape
+    else {
+      MultiShape(layers.map { layer =>
+        layer.build(layer.getInputShape())
+      }.toList)
+    }
+    inputShape
+  }
+
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toMulti()
+    val input1 = input.head.toSingle().toArray
+    if (mergeMode == "concat") {
```

Comment:
let's extract this method into multiple small blocks. i.e: `computOutputShapeForConcat

Reply:
Done.